homeassistant/components/simplisafe/__init__.py

```diff
@@ -117,6 +131,37 @@ def _async_save_refresh_token(hass, config_entry, token):
     )
 
 
+@callback
+def async_create_event_from_raw_data(event_data):
+    """Create a generated payload from raw event data."""
+    event_type = get_event_type_from_payload(event_data)
+
+    # simplisafe-python will take care of logging a message if it finds an event
+    # type it doesn't know about, so if get_event_type_from_payload() returns
+    # None, just return:
+    if not event_type:
+        return
+
+    try:
+        event_sensor_type = EntityTypes(event_data["sensorType"]).name
+    except ValueError:
+        _LOGGER.warning(
+            'Encountered unknown entity type: %s ("%s"). Please report it at'
+            "https://github.com/home-assistant/home-assistant/issues.",
+            event_data["sensorType"],
+            event_data["sensorName"],
+        )
+        event_sensor_type = None
+
+    return {
```

Comment:
Why use a dict and not an `attr` based class ?

Reply:
I think we should now recommend using `dataclasses` based classes instead of `attrs`, unless there's some specific feature of `attrs` that seem needed.

Reply:
I’ll give `dataclass` a shot. 👍🏻